src/context/user.provider.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import {useState, useEffect} from "react"
+import {UserContext} from "./user.context"
+import firebase, {onAuthStateChanged} from "firebase/auth"
+import {auth} from "../utils/firebase/firebase.utils"
+
+type UserProviderProps = {
+	children: React.ReactNode
+}
+
+export const UserProvider: React.FC<UserProviderProps> = (
+	props: UserProviderProps
```

Comment:
De structure props here instead of doing it in the function

Reply:
Done.